src/main/java/org/variantsync/diffdetective/variationtree/VariationNode.java

```diff
@@ -0,0 +1,583 @@
+     */
+    public abstract void setFromLine(int from);
+
+    /**
+     * Returns the end line number of the {@link getLabelLines label} of this node in the
+     * corresponding source code.
+     *
+     * <p>The line number is exclusive (i.e., it points 1 behind the last included line).
+     *
+     * @see setToLine
+     */
+    public abstract int getToLine();
+
+    /**
+     * Changes the line number returned by {@link getToLine}.
+     *
+     * @see getToLine
+     */
+    public abstract void setToLine(int to);
+
+    /**
+     * Returns the range of line numbers of this node's corresponding source code.
+     *
+     * @see getFromLine
+     * @see getToLine
+     */
+    public Lines getLineRange() {
+        return Lines.FromInclToExcl(getFromLine(), getToLine());
+    }
+
+    /**
+     * Returns the parent of this node, or {@code null} if this node doesn't have a parent.
+     *
+     * @see drop
+     * @see addBelow
+     */
+    public abstract VariationNode<T> getParent();
+
+    /**
+     * Returns an unmodifiable list representing the children of this node.
+     *
+     * <p>The following invariant has to hold for all {@code node}s:
+     * <code>
+     *   for (var child : node.getChildren()) {
+     *     Assert.assertTrue(node.isSameAs(child.getParent(node)))
+     *   }
+     * </code>
+     *
+     * @see isChild
+     * @see addChild
+     * @see removeChild
+     */
+    public abstract List<VariationNode<T>> getChildren();
+
+    /**
+     * Returns {@code true} iff this node has no parent.
+     *
+     * @see getParent
+     */
+    public boolean isRoot() {
+        return getParent() == null;
+    }
+
+    /**
+     * Returns {@code true} iff this node has no children.
+     *
+     * @see getChildren
+     */
+    public boolean isLeaf() {
+        return getChildren().isEmpty();
+    }
+
+    /**
+     * Returns the number of child nodes.
+     *
+     * <p>Note: This is O(n) for {@link DiffNode.Projection}.
+     *
+     * @see getChildCount
+     */
+    public int getChildCount() {
+        return getChildren().size();
+    }
+
+    /**
+     * Computes the length of the path from the root node to this node.
+     */
+    public int getDepth() {
+        if (isRoot()) {
+            return 0;
+        }
+
+        return getParent().getDepth() + 1;
+    }
+
+    /**
+     * Computes the number of {@link isAnnotation annotations} of the path from the root node to
+     * this node.
+     */
+    public int getAnnotationDepth() {
+        if (isRoot()) {
+            return 0;
+        }
+
+        if (isIf()) {
+            return getParent().getAnnotationDepth() + 1;
+        }
+
+        return getParent().getAnnotationDepth();
+    }
+
+    /**
+     * Returns the first {@code if} node in the path from this node upwards to the root.
+     */
+    public VariationNode<T> getIfNode() {
+        if (isIf()) {
+            return this;
+        }
+        if (isRoot()) {
+            return null;
```

Comment:
Semantically, this is a dead branch. By our invariant, the root is always an `IF`. We can keep this branch as a security check here but probably its better to throw an Exception instead of returning `null`.

Reply:
You're right this check is redundant. I just remove it and an exception will be thrown if this invariant isn't met (call of a method on a `null` value) which should be enough for an invariant.